ethcore/service/src/service.rs

```diff
@@ -25,19 +25,51 @@ use kvdb::KeyValueDB;
 use kvdb_rocksdb::{Database, DatabaseConfig};
 use stop_guard::StopGuard;
 
-use ethcore::client::{self, Client, ClientConfig, ChainNotify, ClientIoMessage};
-use ethcore::db;
-use ethcore::error::Error;
+use ethsync::PrivateTxHandler;
+use ethcore::client::{Client, ClientConfig, ChainNotify, ClientIoMessage};
+use ethcore::{db, error};
 use ethcore::miner::Miner;
 use ethcore::snapshot::service::{Service as SnapshotService, ServiceParams as SnapServiceParams};
 use ethcore::snapshot::{RestorationStatus};
 use ethcore::spec::Spec;
+use ethcore::account_provider::AccountProvider;
+
+use ethcore_private_tx;
+use Error;
+
+pub struct PrivateTxService {
+	provider: Arc<ethcore_private_tx::Provider>,
+}
+
+impl PrivateTxService {
+	fn new(provider: Arc<ethcore_private_tx::Provider>) -> Self {
+		PrivateTxService {
+			provider,
+		}
+	}
+
+	/// Returns underlying provider.
+	pub fn provider(&self) -> Arc<ethcore_private_tx::Provider> {
```

Comment:
Why some code relies on the internal provider? Not sure I follow the abstractions here I suppose this struct is created to be able to implement the trait from `ethsync`, but I don't see why we make that one public and also the internals public.
Perhaps implement the trait for the entire `Service` instead? This looks like multiplying entities for no good reason.